kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/channel/CommitState.java

```diff
@@ -0,0 +1,167 @@
+          dataWritten.commitId());
+    }
+  }
+
+  void addReady(Envelope envelope) {
+    DataComplete dataComplete = (DataComplete) envelope.event().payload();
+    readyBuffer.add(dataComplete);
+    if (!isCommitInProgress()) {
+      LOG.warn(
+          "Received commit ready when no commit in progress, this can happen during recovery. Commit ID: {}",
+          dataComplete.commitId());
+    }
+  }
+
+  UUID currentCommitId() {
+    return currentCommitId;
+  }
+
+  boolean isCommitInProgress() {
+    return currentCommitId != null;
+  }
+
+  boolean isCommitIntervalReached() {
+    if (startTime == 0) {
+      startTime = System.currentTimeMillis();
+    }
+
+    return (!isCommitInProgress()
+        && System.currentTimeMillis() - startTime >= config.commitIntervalMs());
+  }
+
+  void startNewCommit() {
+    currentCommitId = UUID.randomUUID();
+    startTime = System.currentTimeMillis();
+  }
+
+  void endCurrentCommit() {
+    readyBuffer.clear();
+    currentCommitId = null;
+  }
+
+  void clearResponses() {
+    commitBuffer.clear();
+  }
+
+  boolean isCommitTimedOut() {
+    if (!isCommitInProgress()) {
+      return false;
+    }
+
+    if (System.currentTimeMillis() - startTime > config.commitTimeoutMs()) {
+      LOG.info("Commit timeout reached. Commit ID: {}", currentCommitId);
+      return true;
+    }
+    return false;
+  }
+
+  boolean isCommitReady(int expectedPartitionCount) {
+    if (!isCommitInProgress()) {
+      return false;
+    }
+
+    int receivedPartitionCount =
+        readyBuffer.stream()
+            .filter(payload -> payload.commitId().equals(currentCommitId))
+            .mapToInt(payload -> payload.assignments().size())
+            .sum();
+
+    if (receivedPartitionCount >= expectedPartitionCount) {
+      LOG.info(
+          "Commit {} ready, received responses for all {} partitions",
+          currentCommitId,
+          receivedPartitionCount);
+      return true;
+    }
+
+    LOG.info(
+        "Commit {} not ready, received responses for {} of {} partitions, waiting for more",
+        currentCommitId,
+        receivedPartitionCount,
+        expectedPartitionCount);
+
+    return false;
+  }
+
+  Map<TableReference, List<Envelope>> tableCommitMap() {
+    return commitBuffer.stream()
+        .collect(
+            Collectors.groupingBy(
+                envelope -> ((DataWritten) envelope.event().payload()).tableReference()));
+  }
+
+  OffsetDateTime vtts(boolean partialCommit) {
+    boolean validVtts =
+        !partialCommit
+            && readyBuffer.stream()
+                .flatMap(event -> event.assignments().stream())
+                .allMatch(offset -> offset.timestamp() != null);
+
+    OffsetDateTime result;
+    if (validVtts) {
+      result =
+          readyBuffer.stream()
+              .flatMap(event -> event.assignments().stream())
+              .map(TopicPartitionOffset::timestamp)
+              .min(Comparator.naturalOrder())
+              .get();
```

Comment:
Are we sure this `get()` is safe?  The `allMatch` above will return ture if there are no elements.

Reply:
I changed this to `.orElse(null)`